xDripG5/Transmitter.swift

```diff
@@ -112,10 +121,38 @@ public final class Transmitter: BluetoothManagerDelegate {
 
                         self.log.info("Bonding request sent. Waiting user to respond.")
                     }
-                    let glucose = try peripheral.control(shouldWaitForBond: status.bonded != 0x1)
+
+                    try peripheral.enableNotify(shouldWaitForBond: status.bonded != 0x1)
+                    defer {
+                        peripheral.disconnect()
+                    }
+
+                    let timeMessage = try peripheral.readTimeMessage()
+
+                    let activationDate = Date(timeIntervalSinceNow: -TimeInterval(timeMessage.currentTime))
+
+                    while let command = self.delegate?.dequeuePendingCommand(for: self) {
+                        do {
+                            _ = try peripheral.sendCommand(command, activationDate: activationDate)
+                            self.delegateQueue.async {
```

Comment:
The function of the `didComplete` and `didFail` methods is to allow the commander to make an immediate decision about how to handle subsequent commands, e.g. it might decide to not continue if an earlier command fails.

For this reason, you should call the `didComplete` methods and `didFail` methods from the same queue that `dequeuePendingCommand` is called from.

You should also document this behavior in the TransmitterCommandSource, that its methods are called on a private background queue.

Reply:
Thanks. So, just to clarify, I should create a new `DispatchQueue` (e.g. `commandSourceQueue`) and call all `TransmitterCommandSource` methods on this queue. Is it acceptable to call `dequeuePendingCommand` syncronously; something like this?
```swift
                    while let command: Command = {
                        var c: Command?
                        self.commandSourceQueue.sync {
                            c = self.commandSource?.dequeuePendingCommand(for: self)
                        }
                        return c
                    }() {
```

Reply:
It would work to dispatch to a different queue synchronously, but it doesn't buy anything. I would recommend not creating `commandSourceQueue` and just calling the `commandSource` from the current queue. Clients of `TransmitterCommandSource` just need to know that the methods are called on a "background queue", and it'll be their job to handle thread safety with any memory they choose to access from inside their methods.

Reply:
Thanks. I've added a documentation comment to `TransmitterCommandSource`.